### PR TITLE
feat: Disease filter for collections and datasets must pin 'normal' to the top (#2065)

### PR DESCRIPTION
**Pull request by frano-m**
### Reviewers
**Functional:**

@tihuan 
 
---

## Changes
- added functionality to the filter menu list to display pinned menu items
- disease value "normal" is now pinned as per [Figma](https://www.figma.com/file/18a4QULVIfpFQhTn1FGh8a/Filtering?node-id=3398%3A34024).

### CODE REVIEW

File: frontend/src/common/hooks/useCategoryFilter.ts [hunk 1]
```diff
@@ -1140,7 +1152,7 @@ function isRangeCategoryDisabled(categoryView: RangeCategoryView): boolean {
  */
 function isSelectCategoryDisabled(categoryView: SelectCategoryView): boolean {
   const { isDisabled, values } = categoryView;
-  return isDisabled || values?.every((value) => value.count === 0);
+  return isDisabled || values.every((value) => value.count === 0);
```
Comment (frano-m) [inline frontend/src/common/hooks/useCategoryFilter.ts:1155]: `values` is always defined and so optional chaining is not required.

File: frontend/src/components/common/Filter/components/FilterLabel/components/FilterLabelTooltip/index.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Tooltip } from "czifui";
 import { ReactElement, useMemo } from "react";
-import { categoryTooltipCss } from "src/components/common/Filter/components/FilterLabel/FilterLabelTooltip/style";
+import { categoryTooltipCss } from "src/components/common/Filter/components/FilterLabel/components/FilterLabelTooltip/style";
```
Comment (frano-m) [inline frontend/src/components/common/Filter/components/FilterLabel/components/FilterLabelTooltip/index.tsx:3]: Refactor: folder cleanup - `components` folder is missing

File: frontend/src/components/common/Filter/components/FilterLabel/index.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { Button } from "@blueprintjs/core";
 import { IconNames } from "@blueprintjs/icons";
 import { CATEGORY_LABEL } from "src/components/common/Filter/common/entities";
-import FilterLabelTooltip from "src/components/common/Filter/components/FilterLabel/FilterLabelTooltip";
+import FilterLabelTooltip from "src/components/common/Filter/components/FilterLabel/components/FilterLabelTooltip";
```
Comment (frano-m) [inline frontend/src/components/common/Filter/components/FilterLabel/index.tsx:4]: refactor: `components` folder required

File: frontend/src/components/common/Filter/components/FilterMenu/index.tsx [hunk 4]
```diff
@@ -78,32 +80,64 @@ export default function FilterMenu({
...
+              menuItems={filteredUnpinnedValues}
+              onFilter={onFilter}
+            />
           </MenuItemsScroller>
         )}
       </Menu>
     </MenuWrapper>
   );
 }
+
```
Comment (frano-m) [inline frontend/src/components/common/Filter/components/FilterMenu/index.tsx:116]: refactor: copied from `frontend/src/components/common/Filter/index.tsx`

File: frontend/src/components/common/Filter/index.tsx [hunk 5]
```diff
@@ -188,11 +188,8 @@ function buildSelectCategoryTags(
   categoryView: SelectCategoryView,
   categoryKey: CategoryKey,
   onFilter: OnFilterFn
-): CategoryTag[] | undefined {
+): CategoryTag[] {
   const { values } = categoryView;
-  if (!values) {
```
Comment (frano-m) [inline frontend/src/components/common/Filter/index.tsx:193]: `values` is always defined - truthy check is not required

File: frontend/src/components/common/Filter/components/FilterMenu/index.tsx [hunk 6]
```diff
@@ -78,32 +80,64 @@ export default function FilterMenu({
...
+/**
+ * Returns filtered category values where category key includes search value.
+ * @param values
```
Comment (tihuan) [inline frontend/src/components/common/Filter/components/FilterMenu/index.tsx:119]: discussion: If the params don't have special comments needed, maybe we can drop their JSDoc?

**Review by tihuan — APPROVED** [review]
LGTM!! Thanks so much for adding this 🙏 Only one discussion comment that's not a blocker for merging 😄 🎉 !

File: frontend/src/components/common/Filter/components/FilterMenu/index.tsx (hunk 6 above)
Comment (frano-m) [reply frontend/src/components/common/Filter/components/FilterMenu/index.tsx:119]: Thanks @tihuan I'll fix this now.

**Pull request merged**